TKGame/Entity.cs

```diff
@@ -79,11 +78,17 @@ public Vector2 Size
                         {
                             // Player is above the wall
                             Position.Y -= (int)depth.Y;
+                            if(this is Player)
+                            {
+                                Player.Instance.CollidedVertically = true;
+                                Player.Instance.IsOnGround = true;
+                            }
                         }
                         else
                         {
                             // Player is below the wall
                             Position.Y += (int)depth.Y;
+                            if (this is Player) Player.Instance.CollidedVertically = true;
```

Comment:
So it is both? Be careful how you call this, I can foresee some collision bugs happening if the player manages to go vertically despite this depending on the order things get called.